### PR TITLE
Add Youtube to Blog

### PR DESCRIPTION
**Pull request by glazk0**
Here is my first PR, still working on it, it's just to know if the way I'm doing stuff is good for you.

- Added @tiptap/extension-youtube to blog section
- Added **Delete** button to PostForm
- Added some informations in Blog post

### CODE REVIEW

File: app/dashboard/posts/page.tsx [hunk 1]
```diff
@@ -1,27 +1,49 @@
...
+                </p>
+                {post.published_at && (
+                  <time
+                    dateTime={post.published_at}
+                    className="text-gray-400 text-sm"
+                  >
+                    {format(new Date(post.published_at), 'MM/dd/yyyy')} (
+                    {formatDistance(new Date(post.published_at), new Date(), {
+                      addSuffix: true,
+                    })}
```
Comment (lmachens) [inline app/dashboard/posts/page.tsx:40]: I really like the cards in the admin dashboard, but the dates lead to line-breaks.
Maybe remove the "time ago"?
image

File: app/dashboard/posts/page.tsx [hunk 2]
```diff
@@ -1,27 +1,49 @@
...
+            <Link
+              key={post.id}
+              href={`/dashboard/posts/${post.id}`}
+              className="block py-4"
+            >
+              <div className="flex flex-col h-full rounded-lg border border-gray-700 px-5 py-2.5 text-sm font-medium text-white hover:border-gray-400 focus:outline-none focus:ring-4 focus:ring-[#24292F]/50 disabled:opacity-50 transition-colors">
+                <h2 className="text-lg">{post.title}</h2>
+                <PostHTML className="h-32" html={post.short!} />
+                <p>
+                  Writed by <span className="text-brand-500">{post.user_id!.username ? post.user_id!.username : 'Harry Potter' }</span>
```
Comment (lmachens) [inline app/dashboard/posts/page.tsx:30]: If #10 is merged before this request, this fallback to "Harry Potter" won't be necessary anymore.

**Review by lmachens — CHANGES_REQUESTED** [review]
Good job! It looks much better.
I added some comments.

**Review by lmachens — APPROVED** [review]
Looks good!

File: app/blog/page.tsx [hunk 3]
```diff
@@ -25,23 +29,23 @@ export default async function Page() {
...
-                {formatDistance(new Date(post.published_at), new Date(), {
-                  addSuffix: true,
-                })}
-                )
-              </time>
-            )}
+            <div className="flex">
+              <p className="text-gray-400 text-sm">
+                Writed by{' '}
+                <span className="font-semibold">{post.user_id!.username}</span>{' '}
```
Comment (lmachens) [inline app/blog/page.tsx:40]: This line leads to an error on deployment:

` Type error: Property 'username' does not exist on type 'string & ({ username: unknown; } | { username: unknown; }[])'.` 
Did you face the same issue?

I guess it's the best to check if `user_id` is really an object with `username` .
I am doing something similar in:
https://github.com/lmachens/hogwarts.gg/blob/19cfebb64c4d6cc7d6162e08b6c383da810c2da5/lib/users.ts#L26-L31

**Review by lmachens — CHANGES_REQUESTED** [review]

**Comment by glazk0** [comment]
Let me know if my solution is correct, also thought to maybe add the **User** type later since it's in the #10.

**Review by lmachens — APPROVED** [review]
Good job!
I refactored it a bit and found some missing unique indexes in the database.
With these indexes, we can make sure if an array or single object is returned.
Instead of TS casts, I am using assertions now, which leads to type safety too (but without uncaught errors, if we are wrong).

I renamed the inner joins and prepared a `user` in the posts (we might need need the avatar someday).

**Pull request merged**